pkg/agents/team5/agent.go

```diff
@@ -74,11 +88,137 @@ func (a *CustomAgent5) updateSatisfaction() {
+	}
+	return false
+}
+
+func (a *CustomAgent5) overrideCalculation(treaty messages.Treaty) {
+	if treaty.Request() == messages.LeaveAmountFood {
+		if treaty.RequestOp() == messages.GT && a.CurrPlatFood()-a.attemptFood < food.FoodType(treaty.RequestValue()) {
+			a.attemptFood = food.FoodType(math.Max(0, float64(a.CurrPlatFood()-food.FoodType(treaty.RequestValue())-1)))
+		}
+		if treaty.RequestOp() == messages.GE && a.CurrPlatFood()-a.attemptFood <= food.FoodType(treaty.RequestValue()) {
+			a.attemptFood = food.FoodType(math.Max(0, float64(a.CurrPlatFood()-food.FoodType(treaty.RequestValue()))))
+		}
+	}
+	if treaty.Request() == messages.LeavePercentFood {
+		if treaty.RequestOp() == messages.EQ {
+			a.attemptFood = a.CurrPlatFood() - food.FoodType(float64(a.CurrPlatFood())*float64(treaty.RequestValue())/100)
+		}
+		if treaty.RequestOp() == messages.GT && a.CurrPlatFood()-a.attemptFood < food.FoodType(float64(a.CurrPlatFood())*float64(treaty.RequestValue()/100)) {
+			a.attemptFood = a.CurrPlatFood() - food.FoodType(float64(a.CurrPlatFood())*float64(treaty.RequestValue())/100) - 1
+		}
+		if treaty.RequestOp() == messages.GE && a.CurrPlatFood()-a.attemptFood <= food.FoodType(float64(a.CurrPlatFood())*float64(treaty.RequestValue()/100)) {
+			a.attemptFood = a.CurrPlatFood() - food.FoodType(float64(a.CurrPlatFood())*float64(treaty.RequestValue())/100)
+		}
+	}
+}
+
+func (a *CustomAgent5) treatyOverride() {
+	for _, treaty := range a.ActiveTreaties() {
+		switch {
+		case treaty.Condition() == messages.HP && treaty.ConditionOp() == messages.EQ && a.HP() == treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.HP && treaty.ConditionOp() == messages.GT && a.HP() > treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.HP && treaty.ConditionOp() == messages.GE && a.HP() >= treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.Floor && treaty.ConditionOp() == messages.EQ && a.Floor() == treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.Floor && treaty.ConditionOp() == messages.LT && a.Floor() < treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.Floor && treaty.ConditionOp() == messages.LE && a.Floor() <= treaty.ConditionValue():
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.AvailableFood && treaty.ConditionOp() == messages.EQ && a.CurrPlatFood() == food.FoodType(treaty.ConditionValue()):
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.AvailableFood && treaty.ConditionOp() == messages.GT && a.CurrPlatFood() > food.FoodType(treaty.ConditionValue()):
+			a.overrideCalculation(treaty)
+		case treaty.Condition() == messages.AvailableFood && treaty.ConditionOp() == messages.GE && a.CurrPlatFood() >= food.FoodType(treaty.ConditionValue()):
+			a.overrideCalculation(treaty)
+		}
```

Comment:
TODO: add a fallthrough statement here

Reply:
Please see latest commit to see if resolved